exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/project/ProjectRecordBatch.java

```diff
@@ -136,6 +145,10 @@ public VectorContainer getOutgoingContainer() {
 
   @Override
   protected IterOutcome doWork() {
+    if (wasNone) {
+      return IterOutcome.NONE;
+    }
+
     int incomingRecordCount = incoming.getRecordCount();
 
     if (first && incomingRecordCount == 0) {
```

Comment:
The new logic will handle the case for Project's first outgoing batch. Not sure whether Drill works properly after the first batch getting data and building the schema, but the next incoming batch contains empty result. We may treat as a separate issue for further investigation.